File: app/src/main/java/be/digitalia/fosdem/livedata/SingleEvent.java
package be.digitalia.fosdem.livedata;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Encapsulates data that can only be consumed once.
 */
public class SingleEvent<T> {

	private T content;

	public SingleEvent(@NonNull T content) {
		this.content = content;
	}

	/**
	 * @return The content, or null if it has already been consumed.
	 */
	@Nullable
	public T consume() {
		final T previousContent = content;
		if (previousContent != null) {
			content = null;
		}
		return previousContent;
	}
}

File: app/src/main/java/be/digitalia/fosdem/livedata/SingleEvent.kt
package be.digitalia.fosdem.livedata

/**
 * Encapsulates data that can only be consumed once.
 */
class SingleEvent<T>(content: T) {

    private var content: T? = content

    /**
     * @return The content, or null if it has already been consumed.
     */
    fun consume(): T? {
        val previousContent = content
        if (previousContent != null) {
            content = null
        }
        return previousContent
    }
}